module1/homework/grayscaleImages/compression.hpp

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <array>
+#include <utility>
+#include <vector>
+
+#include <cstdint>
+
+constexpr size_t width = 32;
+constexpr size_t height = 32;
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& getGray);
+std::array<std::array<uint8_t, width>, height> decompressGrayscale(std::vector<std::pair<uint8_t, uint8_t>> Compressed);
+void printMap (const std::array<std::array<uint8_t, width>, height>& getMap);
```

Comment:
You can do this like this :) 
```suggestion
std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& getGray);
using Image = std::array<std::array<uint8_t, width>, height>;

using Compressed = std::vector<std::pair<uint8_t, uint8_t>>;

Compressed compressGrayscale(const Image& compress);

Image decompressGrayscale(const Compressed& compressedMap);

void printMap(const Image& mapToPrint);
```

Reply:
Like Vicx95 is suggesting you should consider making aliases for longer types like std::vector<std::pair<uint8_t, uint8_t>>.

Reply:
Ok, thanks.